cloudify_agent/operations.py

```diff
@@ -411,16 +411,17 @@ def create_agent_amqp(install_agent_timeout=300, manager_ip=None,
     old_agent = _validate_agent()
     _update_broker_config(old_agent, manager_ip, manager_certificate)
     agents = _run_install_script(old_agent, install_agent_timeout)
-    returned_agent = agents['new']
-    ctx.logger.info('Installed agent {0}'.format(returned_agent['name']))
+    new_agent = agents['new']
+    ctx.logger.info('Installed agent {0}'.format(new_agent['name']))
 
     result = _validate_current_amqp()
     if not result['agent_alive']:
         raise RecoverableError('New agent did not start and connect')
 
     # Setting old_cloudify_agent in order to uninstall it later.
     ctx.instance.runtime_properties['old_cloudify_agent'] = agents['old']
-    ctx.instance.runtime_properties['cloudify_agent'] = returned_agent
+    ctx.instance.runtime_properties['cloudify_agent'] = new_agent
+    ctx.instance.update()
```

Comment:
is there a reason for this? at the very end of the operation, there shouldn't be a reason to call update i dont think?

Reply:
fair enough. I get paranoid about this. I'll change it